pyproject.toml

```diff
@@ -90,6 +90,7 @@ youtube-transcript-api = "^0.6.1"
 beautifulsoup4 = "^4.12.2"
 pypdf = "^3.11.0"
 pytube = "^15.0.0"
+elasticsearch = "^8.9.0"
```

Comment:
As always, the question is: do we want to make all dependencies mandatory. I guess, you don't have to answer this question and what you do here is fine.

Reply:
please refer to #407 `setup.py` and `pyproject.toml` to see how this is done if you are unsure.

Reply:
update: using elasticsearch(8.9.0) as extra dependency, can be installed by pip install embedchain[elasticsearch]